### PR TITLE
[TT-7296] fluid checkout integration and latest js components lib version

### PR DESCRIPTION
**Pull request by c5haw**

### CODE REVIEW

**Review by downeyfe — APPROVED** [review]
LGTM, is there a way to add test coverage for this though?

**Comment by c5haw** [comment]
I'll have a think. It might be a bit difficult, but I can see if I can write something that remove the what3words-autosuggest field and triggers the update.

**Comment by c5haw** [comment]
@downeyfe finally got the test working and passing deterministically. Feel free to give a second pass review

File: test/cypress/integration/common/parameters.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { defineParameterType } from 'cypress-cucumber-preprocessor/steps';
+
+defineParameterType({
```
Comment (downeyfe) [inline test/cypress/integration/common/parameters.ts:3]: 👍

File: w3w-autosuggest/public/class-w3w-autosuggest-public.php [hunk 2]
```diff
@@ -258,6 +261,22 @@ public function add_fields_to_checkout_form( $fields ) {
       ];
 
     }
+    // The following doesn't work as it gets reset when WC updates the form in the DOM.
```
Comment (downeyfe) [inline w3w-autosuggest/public/class-w3w-autosuggest-public.php:264]: Is it worth keeping this?

File: w3w-autosuggest/public/js/w3w-autosuggest-public.js [hunk 3]
```diff
@@ -30,50 +32,320 @@
...
+      else $( parent ).prepend(component)
+      components.push(component)
+    }
+
+    return components;
+  }
 
-  function createAutosuggestComponent(target, targetParent, targetSibling) {
+  function generateHiddenInput(name) {
+    const input = document.querySelector(`#${name}`) || document.createElement('input')
```
Comment (downeyfe) [inline w3w-autosuggest/public/js/w3w-autosuggest-public.js:318]: Minor but could use document.getElementId to save building an ID selector

**Pull request merged**